backend/api/quivr_api/modules/knowledge/service/knowledge_service.py

```diff
@@ -23,27 +24,40 @@ def __init__(self, repository: KnowledgeRepository):
         self.repository = repository
 
     async def add_knowledge(
-        self, knowledge_to_add: CreateKnowledgeProperties
+        self,
+        knowledge_to_add: CreateKnowledgeProperties,  # FIXME: (later) @Amine brain id should not be in CreateKnowledgeProperties but since storage is brain_id/file_name
     ) -> Knowledge:
-        knowledge_data = knowledge_to_add.dict()
-        knowledge = KnowledgeDB(**knowledge_data)
+        knowledge = KnowledgeDB(
+            file_name=knowledge_to_add.file_name,
+            url=knowledge_to_add.url,
+            mime_type=knowledge_to_add.mime_type,
+            status=knowledge_to_add.status.value,
+            source=knowledge_to_add.source,
+            source_link=knowledge_to_add.source_link,
+            file_size=knowledge_to_add.file_size,
+            file_sha1=knowledge_to_add.file_sha1,
+            metadata_=knowledge_to_add.metadata,  # type: ignore
+        )
 
         inserted_knowledge_db_instance = await self.repository.insert_knowledge(
             knowledge
         )
 
         assert inserted_knowledge_db_instance.id, "Knowledge ID not generated"
         if inserted_knowledge_db_instance.source == "local":
-            source_link = f"s3://quivr/{inserted_knowledge_db_instance.brain_id}/{inserted_knowledge_db_instance.id}"
-            inserted_knowledge_db_instance.source_link = source_link
+            source_link = f"s3://quivr/{knowledge_to_add.brain_id}/{inserted_knowledge_db_instance.id}"
 
-        inserted_knowledge = await self.repository.insert_knowledge(
-            inserted_knowledge_db_instance
+        inserted_knowledge = await self.repository.update_source_link_knowledge(
+            knowledge_id=inserted_knowledge_db_instance.id, source_link=source_link
+        )
+        new_knowledge_brain = KnowledgeBrain(
```

Comment:
I think that the ORM inserts this automagically when it is related . See many-to-many relationship